nvtabular/io/dataset.py

```diff
@@ -1122,6 +1122,10 @@ def sample_dtypes(self, n=1):
         ``n`` rows of that partition to infer dtype info. If no
         non-empty partitions are found, use the Dask dtypes.
         """
+
+        if hasattr(self.engine, "_dtypes") and self.dtypes is None:
+            return self.engine._dtypes
+
```

Comment:
@jperez999 @karlhigley - This is a temporary hack that I needed for reasonable dask-criteo benchmark performance in google cloud (reading from GCS).  That benchmark made it pretty clear that we want to avoid reading an entire partition from disk to sample the dtypes (which is slow on remote filesystems).

**My plan for a "real" fix in this PR**: I am thinking that a reasonable approach here is to push the `sample_dtypes` logic into `DatasetEngine`.  This way, the disk-backed engines can overrid the implementation with optimized logic to avoid reading an entire partition; for parquet we can only read a single row-group and use cached dtype information if it was already used to calculate `row_groups_per_part`.